ui-admin/src/study/surveys/SurveyView.tsx

```diff
@@ -89,8 +89,13 @@ export const useSurveyParams = () => {
 
 /** routable component for survey editing */
 function SurveyView({ studyEnvContext }: {studyEnvContext: StudyEnvContextT}) {
-  const { isReadOnly, version, stableId } = useSurveyParams()
   const { currentEnv, portal } = studyEnvContext
+
+  if (!currentEnv.studyEnvironmentConfig.initialized) {
+    return <span>Study environment not initialized</span>
+  }
+
+  const { isReadOnly, version, stableId } = useSurveyParams()
```

Comment:
in general, you shouldn't return before a hook (we're somewhat sloppy with this in the code)  See https://stackoverflow.com/questions/54938236/can-you-early-return-with-react-hooks.  

We've been super-sloppy with this in the codebase (you can see we already violated that rule in this component below your edit).  The correct thing here is to make replace the SurveyView component with a useRoutedSurvey hook that contains all the logic that SurveyView has now and returns {isLoading, readOnly, survey}.  
Then have RawSurveyView call the new useRoutedSurvey hook.  Then RawSurveyView will own all the display logic (it will take over rendering the LoadingSpinner).  And then you can rename RawSurveyView to SurveyView.

Reply:
We have a ton of places in our codebase where we have loader/router "components" that really should be refactored into pure hooks.  So as you are updating the codebase, be on the lookout for refactors like this